### ISSUE TITLE
[MRESOLVER-39] deadlock during multithreaded dependency resolution

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resolver#793]
**[Zoltan Haindrich](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kgyrtkirk)** opened **[MRESOLVER-39](https://issues.apache.org/jira/browse/MRESOLVER-39?redirect=false)** and commented

narrowed down a resolution problem to the commit ad50215 ; which belongs to  MRESOLVER-13

Probably the changes in PartialFile might be the most relevant wrt to this problem.

for stack traces, see MNG-6323, MNG-6324

---

**Attachments:**
- [MRESOLVER-39.tar.gz](https://issues.apache.org/jira/secure/attachment/12907406/MRESOLVER-39.tar.gz) (_610 bytes_)

**Issue Links:**
- [MNG-6323](https://issues.apache.org/jira/browse/MNG-6323) Deadlock in multithreaded dependency resolution

- [MRESOLVER-13](https://issues.apache.org/jira/browse/MRESOLVER-13) Exceptions are suppressed incorrectly when closing resources fails
 (_**"is broken by"**_)

1 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Great work! Now looking at it, that commit contains way too much changes :( It looks like the PartialFile changes are some kind of cleanup, but it seems like a nasty detail is missed.

**Comment by jira-importer** [comment]
**[Steve Storey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stevestorey)** commented

I've made a PR that fixes the problem for me: https://github.com/apache/maven-resolver/pull/14 (and comment on the PR includes instructions on how to test / apply to your own local install)

**Comment by jira-importer** [comment]
**[Zoltan Haindrich](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kgyrtkirk)** commented

[Steve Storey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stevestorey) Geat! I can confirm that issue is gone using maven-3.5.2 + pull/14 by using the same things I've used to locate the problem :)

**Comment by jira-importer** [comment]
**[Yegor Borovikov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=yborovikov)** commented

Thank you for hunting it down!

Attaching [^MRESOLVER-39.tar.gz] (3 tiny poms) for easy reproduction:

```java
rm -rf ~/.m2/repository/com/google/code/findbugs/jsr305/3.0.2
mvn -T2 compiler:compile
```

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [599cc2b0f20f373ba5da8a8840d1ae88fafbc2ea](http://git-wip-us.apache.org/repos/asf/maven-resolver/commit/599cc2b0)

**Comment by jira-importer** [comment]
**[Henrik Lundahl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=henrik.lundahl)** commented

Can possibly 1.1.1 be released so that MNG-6323 can be resolved?
